### ISSUE TITLE
Ratchet and Clank TOD [BCES00052]: RSXIO memory not mapped

### ISSUE THREAD

**Issue by Jolty95** [RPCS3/rpcs3#3387]
Hi!
As recomended in #1492 I'm opening this issue with R&C:TOD. 
My config:
-Load automatic and manual: libaudio.sprx (won't boot without libaudio.sprx, but not sound with libaudio.sprx enabled)
-DX12 (using OpenGL or Vulcan makes initial letters looks worse)

Error:
When game it's suppose to show Insomniac logo, it get this error before displaying it.
```
·E 0:03:27.700753 {rsx::thread} HLE: Getting the real address failed. (addr=0x8275e600)
·F 0:03:27.700835 {rsx::thread} class std::runtime_error thrown: GetAddress(offset=0x8275e600, location=0x1): RSXIO memory not mapped
(in file C:\rpcs3\rpcs3\Emu\RSX\RSXThread.cpp:52)
```

Log:
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1273221/RPCS3.log.gz)

Thanks for all!

**Comment by isJuhn** [comment]
This happens because the rsx command buffer being possibly corrupt and the rsx then changes the location of a fragment shader to 0x8275e600. I have yet to discover which thread changes the command buffer and possibly corrupts it.

**Comment by AniLeo** [comment]
·W 0:00:00.000001 Experimental Build Warning! Build origin: RPCS3/rpcs3/master/=3384

In this case 3384 doesn't touch RSX. Otherwise the issue would be invalidated. Always test on master.

**Comment by Jolty95** [comment]
@AniLeo I've downloaded latest from here: https://ci.appveyor.com/project/rpcs3/rpcs3/branch/master :S Sorry if I've done something wrong

**Comment by isJuhn** [comment]
This problem has existed at least since may this year, on every build that tested it.
here is the log from latest master: 
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1273262/RPCS3.log.gz)

**Comment by Asinin3** [comment]
@Jolty95 Always get builds from here, https://rpcs3.net/download the build you were using is basically a PR that hasn't been merged and probably not reviewed for any issues etc. You can use other builds of course but when testing or making an report always use the builds from the main website.

**Comment by Jolty95** [comment]
@Asinin3 Yes, I've found master builds and I apologize for having using a PR instead for testing but error still exist on master too

**Comment by isJuhn** [comment]
Just thought I would update this issue a bit. By masking the address in GetAddress() with 0xFFFFFFF, you get correct graphics and the game can progress through the intro and even through the starting cutscene.
![rpcs3_2017-12-01_20-43-27](https://user-images.githubusercontent.com/26828095/33780056-96690022-dc4f-11e7-8b71-15b9aa8069db.png)
![rpcs3_2017-12-01_20-50-54](https://user-images.githubusercontent.com/26828095/33780076-a8a5abb4-dc4f-11e7-8ebf-b4a136d6881e.png)
however, once it's supposed to go ingame, everything breaks and the RSXthread locks up completely.

Hardware tests are needed, to test what is supposed to happen if you set the address of a fragment shader to something much higher than the normal 256MB.

**Comment by isJuhn** [comment]
Jarves has a [fix](https://github.com/jarveson/rpcs3/commit/5f4455eff7a08aec1481d7842fd3079deb20dca1) for this on his fork. With that commit, the game goes ingame.

**Comment by Sgtbomber91** [comment]
Hello,
Retried BCES00052 (Ratchet and Clank Future: Tools of Destruction) with latest 0.0.5-6975-0ffaf6c1c
Game goes to "Loadable" but ends with 

> GetAddress(offset=0x82b48480, location=0x1): RSXIO memory not mapped

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2120405/RPCS3.log.gz)

**Comment by isJuhn** [comment]
Solved by #5137
